Remove the spack wrapper directories from PATH for the bootstrap step.
This was breaking the build for Cray (and other cross-compile) because
bjam was built for the BE and died on SIGILL on the FE.  See issue
#9613.

This only affects building bjam.  The boost libraries are still built
the normal spack way with the spack wrappers.


diff -Naurb boost_1_66_0.orig/bootstrap.sh boost_1_66_0/bootstrap.sh
--- boost_1_66_0.orig/bootstrap.sh	2017-12-13 17:56:35.000000000 -0600
+++ boost_1_66_0/bootstrap.sh	2019-01-09 13:51:56.407553214 -0600
@@ -7,6 +7,20 @@
 
 # boostinspect:notab - Tabs are required for the Makefile.
 
+NEWPATH=
+OLDIFS="$IFS"
+IFS=:
+
+for dir in $PATH ; do
+   case "x$dir" in
+      *lib*spack*env* ) ;;
+      * ) NEWPATH="${NEWPATH}:${dir}" ;;
+   esac
+done
+
+IFS="$OLDIFS"
+PATH="$NEWPATH"
+
 BJAM=""
 TOOLSET=""
 BJAM_CONFIG=""
